src/common/libsubprocess/client.c

```diff
@@ -146,6 +147,45 @@ flux_future_t *subprocess_rexec (flux_t *h,
     return NULL;
 }
 
+flux_future_t *subprocess_rexec_bg (flux_t *h,
+                                    const char *service_name,
+                                    uint32_t rank,
+                                    const flux_cmd_t *cmd,
+                                    int local_flags)
+{
+    flux_future_t *f = NULL;
+    json_t *ocmd = NULL;
+    char *topic = NULL;
+
+    if (service_name == NULL)
+        service_name = "rexec";
+
+    if (asprintf (&topic, "%s.exec", service_name) < 0) {
+        errno = ENOMEM;
+        goto out;
+    }
+    if (!(ocmd = cmd_tojson (cmd)))
+        goto out;
+
+    /* Clear LOCAL_UNBUF from local_flags. Not a local flag and doesn't
+     * make sense with SUBPROCESS_EXEC_BACKGROUND.
+     */
+    local_flags &= ~FLUX_SUBPROCESS_FLAGS_LOCAL_UNBUF;
+
+    f = flux_rpc_pack (h,
+                       topic,
+                       rank,
+                       FLUX_RPC_STREAMING,
```

Comment:
Should we change the server/protocol to expect FLUX_RPC_STREAMING to not be set with SUBPROCESS_EXEC_BACKGROUND?

It sounds like the user is expected to not treat the RPC as streaming and will never be waiting for the ENODATA, which would not be required to send if the STREAMING flag weren't set in the request.

Reply:
Yeah, I considered this too and figured the question would come up. I wondered if just having a separate RPC instead of a flag would be more appropriate here. Especially with the special-case handling of the other flags (most are ignored), etc.